editors/code/src/toolchain.ts

```diff
@@ -139,35 +140,36 @@ export async function getRustcId(dir: string): Promise<string> {
 }
 
 /** Mirrors `toolchain::cargo()` implementation */
-export function cargoPath(): string {
+export function cargoPath(): Promise<string> {
     return getPathForExecutable("cargo");
 }
 
 /** Mirrors `toolchain::get_path_for_executable()` implementation */
-export const getPathForExecutable = memoize(
+export const getPathForExecutable = memoizeAsync(
     // We apply caching to decrease file-system interactions
-    (executableName: "cargo" | "rustc" | "rustup"): string => {
+    async (executableName: "cargo" | "rustc" | "rustup"): Promise<string> => {
         {
             const envVar = process.env[executableName.toUpperCase()];
             if (envVar) return envVar;
         }
 
-        if (lookupInPath(executableName)) return executableName;
+        if (await lookupInPath(executableName)) return executableName;
 
         try {
             // hmm, `os.homedir()` seems to be infallible
             // it is not mentioned in docs and cannot be infered by the type signature...
             const standardPath = vscode.Uri.joinPath(vscode.Uri.file(os.homedir()), ".cargo", "bin", executableName);
 
-            if (isFileAtUri(standardPath)) return standardPath.fsPath;
+            const exist = await isFileAtUri(standardPath);
+            if (exist) return standardPath.fsPath;
```

Comment:
Personally, I think this is a little verbose, and it's inconsistent with line 156.

Reply:
done